### PR TITLE
emit cluster creation duration metric

### PR DESCRIPTION
**Pull request by xh3b4sd**
As a next step to improve our status metrics we want to emit cluster creation durations. The duration computation in this proposal is done in our go collector. Ross and I didn't figure how to let Prometheus do that computation. It was also unclear how multiple update durations would be tracked later.

### CODE REVIEW

File: status_collector.go [hunk 1]
```diff
@@ -1,133 +0,0 @@
-package statusresource
```
Comment (xh3b4sd) [inline status_collector.go]: We renamed the file. That is probably why the diff is not useful.

File: legacy_status_collector.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+				t1 := timeForCreatedCondition(p)
+				t2 := m.GetCreationTimestamp().Time
+
+				ch <- prometheus.MustNewConstMetric(
+					legacyCreationSecondsCollectorDescription,
+					prometheus.GaugeValue,
+					t1.Sub(t2).Seconds(),
+					m.GetName(),
+				)
+			}
```
Comment (xh3b4sd) [inline legacy_status_collector.go]: This block is new.

File: legacy_status_collector.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	legacyStatusCollectorDescription *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("statusresource", "cluster", "status"),
+		"Cluster status condition as provided by the CR status.",
+		[]string{
+			"cluster_id",
+			"status",
+		},
+		nil,
+	)
+	legacyCreationSecondsCollectorDescription *prometheus.Desc = prometheus.NewDesc(
```
Comment (xh3b4sd) [inline legacy_status_collector.go]: This metric is new.

**Review by xh3b4sd — COMMENTED** [review]
Note that I still have to test this.

File: legacy_status_collector.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+			}
+
+			if p.ClusterStatus().HasCreatedCondition() {
+				t1 := timeForCreatedCondition(p)
+				t2 := m.GetCreationTimestamp().Time
+
+				ch <- prometheus.MustNewConstMetric(
+					legacyCreationSecondsCollectorDescription,
+					prometheus.GaugeValue,
+					t1.Sub(t2).Seconds(),
```
Comment (tomahawk28) [inline legacy_status_collector.go]: Could it be summary or histogram to see how it distributed?

**Review by tomahawk28 — APPROVED** [review]

File: legacy_status_collector.go (hunk 4 above)
Comment (xh3b4sd) [reply legacy_status_collector.go]: I think a histogram only makes sense for changing values. The creation time does never change once emitted.

Comment (tomahawk28) [reply legacy_status_collector.go]: Thanks, that make sense.

**Comment by xh3b4sd** [comment]
Tested on `godsmack`. The cluster creation took about 10 minutes. 

Screenshot 2019-05-27 at 14 28 28

**Pull request merged**